90zfsbootmenu/zfsbootmenu-lib.sh

```diff
@@ -253,20 +263,22 @@ draw_kernel() {
 # returns: 130 on error, 0 otherwise
 
 draw_snapshots() {
-  local benv selected header expects
+  local benv selected header expects sort_key
 
   benv="${1}"
 
   zdebug "using boot environment: ${benv}"
 
+  sort_key="$( get_sort_key )"
+
   header="$( header_wrap \
     "[ENTER] duplicate" "[ESC] back" "[CTRL+H] help" "" \
     "[CTRL+X] clone and promote" "[CTRL+C] clone only" "" \
     "[CTRL+I] interactive chroot" "[CTRL+D] show diff" )"
 
-  expects="--expect=alt-x,alt-c,alt-d,alt-i"
+  expects="--expect=alt-x,alt-c,alt-d,alt-i,alt-o"
 
-  if ! selected="$( zfs list -t snapshot -H -o name "${benv}" |
```

Comment:
If the input is always sanitized, this is fine. Alternatively, what do you think about a `zfs_list_sorted` function that runs `zfs list -S "$( get_sort_key )" "@"` and prints the output on success, otherwise falls back to `zfs list "@"` in failure?

Reply:
There are two places currently where that'd be helpful, `populate_be_list()` and `draw_snapshots()`.  I'm not sure the added complexity is worth it for that, but I'm not opposed to it either.

Reply:
I don't feel strongly about it. The main value would be in allowing arbitrary user fields without verifying them against an approved list. As long as there is a pre-approved list, the behavior should be correct without a fallback.

Reply:
I think I'm going to punt on this for now.